libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamOutput.java

```diff
@@ -495,6 +497,14 @@ public final void writeBigInteger(BigInteger v) throws IOException {
         writeString(v.toString());
     }
 
+    public final void writeMediaType(final MediaType v) throws IOException {
+        if (v.getClass().isEnum()) {
```

Comment:
Why this constraint? Seems a bit counter to object oriented principles to require that a particular interface be implemented as an enum.

Reply:
That method isn't used. I forgot to delete it ;)